[EISW-141730] 19.x Merge

rayngun opened the pull request:
## Summary
Apply custom VPUx-Plugin custom changes in npu/release/18.x to npu/release/19.x. Some cherry-picks did not apply cleanly or caused runtime issues. Fixes to the cherry-picks are squashed back into original commit so some commits will look a little different from npu/release/18.x. In addition, the following custom changes were also done. JIRA tickets are created to track work needed to remove the custom changes.

* XeGPU disabled (https://jira.devtools.intel.com/browse/EISW-163563)
* TosaToBuild disabled (https://jira.devtools.intel.com/browse/EISW-163564)
* OutputTextStrategy exposed in Timing module (https://jira.devtools.intel.com/browse/EISW-163565)
* Removed SymbolTable trait check (https://jira.devtools.intel.com/browse/EISW-138682)

## JIRA ticket

* [EISW-141730](https://jira.devtools.intel.com/browse/EISW-141730)

## Related PR in NPU Compiler and/or OpenVINO repository with sub-module update

* PR-xxx

### Other related tickets
> List tickets for additional work, eg, something was found during review but you agreed to address it in another Jira

* E-xxxxx

Commit:
[mlir][RFC] Allow to skip operands type specification in ASM form

Based on the following discussion:
https://llvm.discourse.group/t/declarative-assembly-format-requirement-for-type-presence/4399

Relax checks in `OperationParser` - it allows to skip value type specification,
if the value was already defined in the same block.

Differential Revision: https://reviews.llvm.org/D111650

nikita-kud committed:
Add FlatSymbolRefAttr and ArrayAttr for memory space representation (#44/#45)

Commit:
[mlir] Relax ASM parser requirements about operands Type presence

Commit:
[mlir] Remove LIT tests affected by ASM simplification

Jackojc committed:
Add flag to disable unused GPU dialects in MLIR

nikita-kud committed:
Merge pull request #11 from iszilve/SymbolTableTweaks

add support for SymbolContainer trait in MLIR infrastructure

Commit:
extend SymbolContainer semantics to SymbolUserMap

nikita-kud committed:
[MLIR] Extend MlirOptMain API to register HW-specific options inside

nikita-kud committed:
Update PR template

See PR22 for details

nikita-kud committed:
Use virtual inheritance for PassPipelineOptions

See PR25 for details

nikita-kud committed:
Add method getInlineBlockAndPoint for DialectInlinerInterface

nikita-kud committed:
Disable failed tests: #73140

xin-zhang-intel committed:
PR #38: Replace BaseMemRef/TensorType class with TypeInterface

Jacenty-And-Intel committed:
PR #27: Adding newlines and tracking indent level moved to mlir::AsmPrinter

AndreiFilipIntel committed:
Added f8E4M3FN and f8E5M2 support to existing QuantizedTypes (17.x) (#37)

* added integral type check
* changed to f8e_m_ format, removed isF8 helper method, formatting
* added negative tests, removed duplicate test
* Added endline in parse-uniform-invalid.mlir
* changed to f8E5M2/f8E4M3FN format, enabled default type parsing
* updated parse-any-invalid error message checks

andrey-golubev committed:
Customize memref->tensor conversion via dialect interface (#40)

Add a special DialectBufferizerInterface that handles
memref -> tensor conversions. Custom dialects must provide the
interface implementation for the tensor/memref types they expose.

Provide an implementation for Builtin dialect so that
mlir::*MemRefType -> mlir::*TensorType is supported natively. However,
do this in a form of interface "extension" so that the users could still
overwrite the default behavior (e.g. to customize MemRefType::layout ->
RankedTensorType::encoding conversion).

Use an interface wrapper in memref::getTensorTypeFromMemRefType() - the
main place that the one-shot bufferization relies on for type inference,
type checking and so on.

nikita-kud committed:
Initialize pass options using pipeline options

jschj committed:
Fix compilation issue with MSVC

Maxim-Doronin committed:
SDL security fixes (#32)

* Bump the llvm-docs-requirements group in /llvm/docs with 23 updates

Bumps the llvm-docs-requirements group in /llvm/docs with 23 updates:

| Package | From | To |
| --- | --- | --- |
| [alabaster](https://github.com/sphinx-doc/alabaster) | `0.7.13` | `0.7.16` |
| [babel](https://github.com/python-babel/babel) | `2.14.0` | `2.15.0` |
| [beautifulsoup4](https://www.crummy.com/software/BeautifulSoup/bs4/) | `4.12.2` | `4.12.3` |
| [certifi](https://github.com/certifi/python-certifi) | `2023.11.17` | `2024.6.2` |
| [docutils](https://docutils.sourceforge.io) | `0.20.1` | `0.21.2` |
| [furo](https://github.com/pradyunsg/furo) | `2023.8.19` | `2024.5.6` |
| [idna](https://github.com/kjd/idna) | `3.6` | `3.7` |
| [jinja2](https://github.com/pallets/jinja) | `3.1.2` | `3.1.4` |
| [markdown](https://github.com/Python-Markdown/markdown) | `3.5.1` | `3.6` |
| [markupsafe](https://github.com/pallets/markupsafe) | `2.1.3` | `2.1.5` |
| [mdit-py-plugins](https://github.com/executablebooks/mdit-py-plugins) | `0.4.0` | `0.4.1` |
| [myst-parser](https://github.com/executablebooks/MyST-Parser) | `2.0.0` | `3.0.1` |
| [packaging](https://github.com/pypa/packaging) | `23.2` | `24.0` |
| [pygments](https://github.com/pygments/pygments) | `2.17.2` | `2.18.0` |
| [requests](https://github.com/psf/requests) | `2.31.0` | `2.32.3` |
| [sphinx](https://github.com/sphinx-doc/sphinx) | `7.1.2` | `7.3.7` |
| [sphinx-automodapi](https://github.com/astropy/sphinx-automodapi) | `0.16.0` | `0.17.0` |
| [sphinxcontrib-applehelp](https://github.com/sphinx-doc/sphinxcontrib-applehelp) | `1.0.4` | `1.0.8` |
| [sphinxcontrib-devhelp](https://github.com/sphinx-doc/sphinxcontrib-devhelp) | `1.0.5` | `1.0.6` |
| [sphinxcontrib-htmlhelp](https://github.com/sphinx-doc/sphinxcontrib-htmlhelp) | `2.0.4` | `2.0.5` |
| [sphinxcontrib-qthelp](https://github.com/sphinx-doc/sphinxcontrib-qthelp) | `1.0.6` | `1.0.7` |
| [sphinxcontrib-serializinghtml](https://github.com/sphinx-doc/sphinxcontrib-serializinghtml) | `1.1.9` | `1.1.10` |
| [urllib3](https://github.com/urllib3/urllib3) | `2.1.0` | `2.2.1` |

Updates `alabaster` from 0.7.13 to 0.7.16
- [Release notes](https://github.com/sphinx-doc/alabaster/releases)
- [Changelog](https://github.com/sphinx-doc/alabaster/blob/master/docs/changelog.rst)
- [Commits](https://github.com/sphinx-doc/alabaster/compare/0.7.13...0.7.16)

Updates `babel` from 2.14.0 to 2.15.0
- [Release notes](https://github.com/python-babel/babel/releases)
- [Changelog](https://github.com/python-babel/babel/blob/master/CHANGES.rst)
- [Commits](https://github.com/python-babel/babel/compare/v2.14.0...v2.15.0)

Updates `beautifulsoup4` from 4.12.2 to 4.12.3

Updates `certifi` from 2023.11.17 to 2024.6.2
- [Commits](https://github.com/certifi/python-certifi/compare/2023.11.17...2024.06.02)

Updates `docutils` from 0.20.1 to 0.21.2

Updates `furo` from 2023.8.19 to 2024.5.6
- [Release notes](https://github.com/pradyunsg/furo/releases)
- [Changelog](https://github.com/pradyunsg/furo/blob/main/docs/changelog.md)
- [Commits](https://github.com/pradyunsg/furo/compare/2023.08.19...2024.05.06)

Updates `idna` from 3.6 to 3.7
- [Release notes](https://github.com/kjd/idna/releases)
- [Changelog](https://github.com/kjd/idna/blob/master/HISTORY.rst)
- [Commits](https://github.com/kjd/idna/compare/v3.6...v3.7)

Updates `jinja2` from 3.1.2 to 3.1.4
- [Release notes](https://github.com/pallets/jinja/releases)
- [Changelog](https://github.com/pallets/jinja/blob/main/CHANGES.rst)
- [Commits](https://github.com/pallets/jinja/compare/3.1.2...3.1.4)

Updates `markdown` from 3.5.1 to 3.6
- [Release notes](https://github.com/Python-Markdown/markdown/releases)
- [Changelog](https://github.com/Python-Markdown/markdown/blob/master/docs/changelog.md)
- [Commits](https://github.com/Python-Markdown/markdown/compare/3.5.1...3.6)

Updates `markupsafe` from 2.1.3 to 2.1.5
- [Release notes](https://github.com/pallets/markupsafe/releases)
- [Changelog](https://github.com/pallets/markupsafe/blob/main/CHANGES.rst)
- [Commits](https://github.com/pallets/markupsafe/compare/2.1.3...2.1.5)

Updates `mdit-py-plugins` from 0.4.0 to 0.4.1
- [Release notes](https://github.com/executablebooks/mdit-py-plugins/releases)
- [Changelog](https://github.com/executablebooks/mdit-py-plugins/blob/master/CHANGELOG.md)
- [Commits](https://github.com/executablebooks/mdit-py-plugins/compare/v0.4.0...v0.4.1)

Updates `myst-parser` from 2.0.0 to 3.0.1
- [Release notes](https://github.com/executablebooks/MyST-Parser/releases)
- [Changelog](https://github.com/executablebooks/MyST-Parser/blob/master/CHANGELOG.md)
- [Commits](https://github.com/executablebooks/MyST-Parser/compare/v2.0.0...v3.0.1)

Updates `packaging` from 23.2 to 24.0
- [Release notes](https://github.com/pypa/packaging/releases)
- [Changelog](https://github.com/pypa/packaging/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pypa/packaging/compare/23.2...24.0)

Updates `pygments` from 2.17.2 to 2.18.0
- [Release notes](https://github.com/pygments/pygments/releases)
- [Changelog](https://github.com/pygments/pygments/blob/master/CHANGES)
- [Commits](https://github.com/pygments/pygments/compare/2.17.2...2.18.0)

Updates `requests` from 2.31.0 to 2.32.3
- [Release notes](https://github.com/psf/requests/releases)
- [Changelog](https://github.com/psf/requests/blob/main/HISTORY.md)
- [Commits](https://github.com/psf/requests/compare/v2.31.0...v2.32.3)

Updates `sphinx` from 7.1.2 to 7.3.7
- [Release notes](https://github.com/sphinx-doc/sphinx/releases)
- [Changelog](https://github.com/sphinx-doc/sphinx/blob/master/CHANGES.rst)
- [Commits](https://github.com/sphinx-doc/sphinx/compare/v7.1.2...v7.3.7)

Updates `sphinx-automodapi` from 0.16.0 to 0.17.0
- [Release notes](https://github.com/astropy/sphinx-automodapi/releases)
- [Changelog](https://github.com/astropy/sphinx-automodapi/blob/main/CHANGES.rst)
- [Commits](https://github.com/astropy/sphinx-automodapi/compare/v0.16.0...v0.17.0)

Updates `sphinxcontrib-applehelp` from 1.0.4 to 1.0.8
- [Release notes](https://github.com/sphinx-doc/sphinxcontrib-applehelp/releases)
- [Changelog](https://github.com/sphinx-doc/sphinxcontrib-applehelp/blob/master/CHANGES)
- [Commits](https://github.com/sphinx-doc/sphinxcontrib-applehelp/compare/1.0.4...1.0.8)

Updates `sphinxcontrib-devhelp` from 1.0.5 to 1.0.6
- [Release notes](https://github.com/sphinx-doc/sphinxcontrib-devhelp/releases)
- [Changelog](https://github.com/sphinx-doc/sphinxcontrib-devhelp/blob/1.0.6/CHANGES)
- [Commits](https://github.com/sphinx-doc/sphinxcontrib-devhelp/compare/1.0.5...1.0.6)

Updates `sphinxcontrib-htmlhelp` from 2.0.4 to 2.0.5
- [Release notes](https://github.com/sphinx-doc/sphinxcontrib-htmlhelp/releases)
- [Changelog](https://github.com/sphinx-doc/sphinxcontrib-htmlhelp/blob/2.0.5/CHANGES)
- [Commits](https://github.com/sphinx-doc/sphinxcontrib-htmlhelp/compare/2.0.4...2.0.5)

Updates `sphinxcontrib-qthelp` from 1.0.6 to 1.0.7
- [Release notes](https://github.com/sphinx-doc/sphinxcontrib-qthelp/releases)
- [Changelog](https://github.com/sphinx-doc/sphinxcontrib-qthelp/blob/1.0.7/CHANGES)
- [Commits](https://github.com/sphinx-doc/sphinxcontrib-qthelp/compare/1.0.6...1.0.7)

Updates `sphinxcontrib-serializinghtml` from 1.1.9 to 1.1.10
- [Release notes](https://github.com/sphinx-doc/sphinxcontrib-serializinghtml/releases)
- [Changelog](https://github.com/sphinx-doc/sphinxcontrib-serializinghtml/blob/master/CHANGES)
- [Commits](https://github.com/sphinx-doc/sphinxcontrib-serializinghtml/compare/1.1.9...1.1.10)

Updates `urllib3` from 2.1.0 to 2.2.1
- [Release notes](https://github.com/urllib3/urllib3/releases)
- [Changelog](https://github.com/urllib3/urllib3/blob/main/CHANGES.rst)
- [Commits](https://github.com/urllib3/urllib3/compare/2.1.0...2.2.1)

---
updated-dependencies:
- dependency-name: alabaster
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: llvm-docs-requirements
- dependency-name: babel
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: llvm-docs-requirements
- dependency-name: beautifulsoup4
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: llvm-docs-requirements
- dependency-name: certifi
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: llvm-docs-requirements
- dependency-name: docutils
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: llvm-docs-requirements
- dependency-name: furo
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: llvm-docs-requirements
- dependency-name: idna
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: llvm-docs-requirements
- dependency-name: jinja2
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: llvm-docs-requirements
- dependency-name: markdown
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: llvm-docs-requirements
- dependency-name: markupsafe
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: llvm-docs-requirements
- dependency-name: mdit-py-plugins
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: llvm-docs-requirements
- dependency-name: myst-parser
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: llvm-docs-requirements
- dependency-name: packaging
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: llvm-docs-requirements
- dependency-name: pygments
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: llvm-docs-requirements
- dependency-name: requests
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: llvm-docs-requirements
- dependency-name: sphinx
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: llvm-docs-requirements
- dependency-name: sphinx-automodapi
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: llvm-docs-requirements
- dependency-name: sphinxcontrib-applehelp
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: llvm-docs-requirements
- dependency-name: sphinxcontrib-devhelp
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: llvm-docs-requirements
- dependency-name: sphinxcontrib-htmlhelp
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: llvm-docs-requirements
- dependency-name: sphinxcontrib-qthelp
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: llvm-docs-requirements
- dependency-name: sphinxcontrib-serializinghtml
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: llvm-docs-requirements
- dependency-name: urllib3
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: llvm-docs-requirements
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump the github-actions group with 8 updates

Bumps the github-actions group with 8 updates:

| Package | From | To |
| --- | --- | --- |
| [tj-actions/changed-files](https://github.com/tj-actions/changed-files) | `39` | `44` |
| [actions/setup-python](https://github.com/actions/setup-python) | `4` | `5` |
| [actions/download-artifact](https://github.com/actions/download-artifact) | `3` | `4` |
| [actions/setup-node](https://github.com/actions/setup-node) | `3` | `4` |
| [actions/github-script](https://github.com/actions/github-script) | `6` | `7` |
| [actions/labeler](https://github.com/actions/labeler) | `4` | `5` |
| [ossf/scorecard-action](https://github.com/ossf/scorecard-action) | `2.1.2` | `2.3.3` |
| [github/codeql-action](https://github.com/github/codeql-action) | `2.2.4` | `3.25.8` |

Updates `tj-actions/changed-files` from 39 to 44
- [Release notes](https://github.com/tj-actions/changed-files/releases)
- [Changelog](https://github.com/tj-actions/changed-files/blob/main/HISTORY.md)
- [Commits](https://github.com/tj-actions/changed-files/compare/v39...v44)

Updates `actions/setup-python` from 4 to 5
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v4...v5)

Updates `actions/download-artifact` from 3 to 4
- [Release notes](https://github.com/actions/download-artifact/releases)
- [Commits](https://github.com/actions/download-artifact/compare/v3...v4)

Updates `actions/setup-node` from 3 to 4
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v3...v4)

Updates `actions/github-script` from 6 to 7
- [Release notes](https://github.com/actions/github-script/releases)
- [Commits](https://github.com/actions/github-script/compare/v6...v7)

Updates `actions/labeler` from 4 to 5
- [Release notes](https://github.com/actions/labeler/releases)
- [Commits](https://github.com/actions/labeler/compare/v4...v5)

Updates `ossf/scorecard-action` from 2.1.2 to 2.3.3
- [Release notes](https://github.com/ossf/scorecard-action/releases)
- [Changelog](https://github.com/ossf/scorecard-action/blob/main/RELEASE.md)
- [Commits](https://github.com/ossf/scorecard-action/compare/e38b1902ae4f44df626f11ba0734b14fb91f8f86...dc50aa9510b46c811795eb24b2f1ba02a914e534)

Updates `github/codeql-action` from 2.2.4 to 3.25.8
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/17573ee1cc1b9d061760f3a006fc4aac4f944fd5...2e230e8fe0ad3a14a340ad0815ddb96d599d2aff)

---
updated-dependencies:
- dependency-name: tj-actions/changed-files
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: actions/download-artifact
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: actions/setup-node
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: actions/github-script
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: actions/labeler
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: ossf/scorecard-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: github-actions
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump black from 23.9.1 to 24.3.0 in /llvm/utils/git

Bumps [black](https://github.com/psf/black) from 23.9.1 to 24.3.0.
- [Release notes](https://github.com/psf/black/releases)
- [Changelog](https://github.com/psf/black/blob/main/CHANGES.md)
- [Commits](https://github.com/psf/black/compare/23.9.1...24.3.0)

---
updated-dependencies:
- dependency-name: black
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump requests from 2.31.0 to 2.32.2 in /llvm/utils/git

Bumps [requests](https://github.com/psf/requests) from 2.31.0 to 2.32.2.
- [Release notes](https://github.com/psf/requests/releases)
- [Changelog](https://github.com/psf/requests/blob/main/HISTORY.md)
- [Commits](https://github.com/psf/requests/compare/v2.31.0...v2.32.2)

---
updated-dependencies:
- dependency-name: requests
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump cryptography from 41.0.3 to 42.0.4 in /llvm/utils/git

Bumps [cryptography](https://github.com/pyca/cryptography) from 41.0.3 to 42.0.4.
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/41.0.3...42.0.4)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump urllib3 from 2.0.4 to 2.0.7 in /llvm/utils/git

Bumps [urllib3](https://github.com/urllib3/urllib3) from 2.0.4 to 2.0.7.
- [Release notes](https://github.com/urllib3/urllib3/releases)
- [Changelog](https://github.com/urllib3/urllib3/blob/main/CHANGES.rst)
- [Commits](https://github.com/urllib3/urllib3/compare/2.0.4...2.0.7)

---
updated-dependencies:
- dependency-name: urllib3
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump idna from 3.4 to 3.7 in /llvm/utils/git

Bumps [idna](https://github.com/kjd/idna) from 3.4 to 3.7.
- [Release notes](https://github.com/kjd/idna/releases)
- [Changelog](https://github.com/kjd/idna/blob/master/HISTORY.rst)
- [Commits](https://github.com/kjd/idna/compare/v3.4...v3.7)

---
updated-dependencies:
- dependency-name: idna
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Maxim-Doronin committed:
SDL fixes 2 (#43)

* Bump braces from 3.0.2 to 3.0.3 in /mlir/utils/vscode

Bumps [braces](https://github.com/micromatch/braces) from 3.0.2 to 3.0.3.
- [Changelog](https://github.com/micromatch/braces/blob/master/CHANGELOG.md)
- [Commits](https://github.com/micromatch/braces/compare/3.0.2...3.0.3)

---
updated-dependencies:
- dependency-name: braces
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump packaging

Bumps the llvm-docs-requirements group with 1 update in the /llvm/docs directory: [packaging](https://github.com/pypa/packaging).

Updates `packaging` from 24.0 to 24.1
- [Release notes](https://github.com/pypa/packaging/releases)
- [Changelog](https://github.com/pypa/packaging/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pypa/packaging/compare/24.0...24.1)

---
updated-dependencies:
- dependency-name: packaging
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: llvm-docs-requirements
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump the github-actions group across 1 directory with 2 updates

Bumps the github-actions group with 2 updates in the / directory: [actions/checkout](https://github.com/actions/checkout) and [github/codeql-action](https://github.com/github/codeql-action).

Updates `actions/checkout` from 3.1.0 to 4.1.7
- [Release notes](https://github.com/actions/checkout/releases)
- [Commits](https://github.com/actions/checkout/compare/v3.1.0...v4.1.7)

Updates `github/codeql-action` from 3.25.8 to 3.25.10
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/2e230e8fe0ad3a14a340ad0815ddb96d599d2aff...23acc5c183826b7a8a97bce3cecc52db901f8251)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump urllib3 from 2.0.7 to 2.2.2 in /llvm/utils/git

Bumps [urllib3](https://github.com/urllib3/urllib3) from 2.0.7 to 2.2.2.
- [Release notes](https://github.com/urllib3/urllib3/releases)
- [Changelog](https://github.com/urllib3/urllib3/blob/main/CHANGES.rst)
- [Commits](https://github.com/urllib3/urllib3/compare/2.0.7...2.2.2)

---
updated-dependencies:
- dependency-name: urllib3
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump urllib3 from 2.2.1 to 2.2.2 in /llvm/docs

Bumps [urllib3](https://github.com/urllib3/urllib3) from 2.2.1 to 2.2.2.
- [Release notes](https://github.com/urllib3/urllib3/releases)
- [Changelog](https://github.com/urllib3/urllib3/blob/main/CHANGES.rst)
- [Commits](https://github.com/urllib3/urllib3/compare/2.2.1...2.2.2)

---
updated-dependencies:
- dependency-name: urllib3
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump minimatch from 3.0.4 to 3.1.2 in /mlir/utils/vscode

Bumps [minimatch](https://github.com/isaacs/minimatch) from 3.0.4 to 3.1.2.
- [Changelog](https://github.com/isaacs/minimatch/blob/main/changelog.md)
- [Commits](https://github.com/isaacs/minimatch/compare/v3.0.4...v3.1.2)

---
updated-dependencies:
- dependency-name: minimatch
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump semver from 5.7.1 to 5.7.2 in /llvm/utils/vscode/llvm

Bumps [semver](https://github.com/npm/node-semver) from 5.7.1 to 5.7.2.
- [Release notes](https://github.com/npm/node-semver/releases)
- [Changelog](https://github.com/npm/node-semver/blob/v5.7.2/CHANGELOG.md)
- [Commits](https://github.com/npm/node-semver/compare/v5.7.1...v5.7.2)

---
updated-dependencies:
- dependency-name: semver
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump gitpython from 3.1.32 to 3.1.41 in /llvm/utils/git

Bumps [gitpython](https://github.com/gitpython-developers/GitPython) from 3.1.32 to 3.1.41.
- [Release notes](https://github.com/gitpython-developers/GitPython/releases)
- [Changelog](https://github.com/gitpython-developers/GitPython/blob/main/CHANGES)
- [Commits](https://github.com/gitpython-developers/GitPython/compare/3.1.32...3.1.41)

---
updated-dependencies:
- dependency-name: gitpython
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump qs from 6.10.1 to 6.12.1 in /mlir/utils/vscode

Bumps [qs](https://github.com/ljharb/qs) from 6.10.1 to 6.12.1.
- [Changelog](https://github.com/ljharb/qs/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ljharb/qs/compare/v6.10.1...v6.12.1)

---
updated-dependencies:
- dependency-name: qs
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

* cryptography version is bumped

* Bump urllib3 version

* Bump requests version

* Bump idna version

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

DariaMityagina committed:
Port Ubunu-24 warning fixes to npu/release/18.x (#45)

* Fix <anonymous> declared here

* Fix Ubuntu24 (#33)

Natan-GabrielTiutiuIntel committed:
[mlir] Add --list-passes option to mlir-opt (#100420) (#52)

Currently, the only way to see the passes that were registered is by
calling “mlir-opt --help”. However, for compilers with 500+ passes, the
help message becomes too long and sometimes hard to understand. In this
PR I add a new "--list-passes" option to mlir-opt, which can be used for
printing only the registered passes, a feature that would be extremely
useful.

sartil committed:
Extend Quant dialect with Quantile Quantization type (#53)

* Expanding Quant dialect with Quantile Quantized type

* Adding quantile mlir tests

* Adding check on quantiles array size and updated mlir tests

dependabot[bot] committed:
Bump certifi from 2024.6.2 to 2024.7.4 in /llvm/docs (#48)

Bumps [certifi](https://github.com/certifi/python-certifi) from 2024.6.2 to 2024.7.4.
- [Commits](https://github.com/certifi/python-certifi/compare/2024.06.02...2024.07.04)

---
updated-dependencies:
- dependency-name: certifi
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump certifi from 2023.7.22 to 2024.7.4 in /llvm/utils/git (#47)

Bumps [certifi](https://github.com/certifi/python-certifi) from 2023.7.22 to 2024.7.4.
- [Commits](https://github.com/certifi/python-certifi/compare/2023.07.22...2024.07.04)

---
updated-dependencies:
- dependency-name: certifi
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ArtemySkrebkov committed:
Disable TosaToTensor

ArtemySkrebkov committed:
Disable XeGPU

sartil committed:
Extending QuantileQuantizedType with quantileType mlir::Type member (#60)

nikita-kud committed:
[ADT] Require base equality in indexed_accessor_iterator::operator==() (#107856) (#64)

Similarly to operator<(), equality-comparing iterators from different
ranges must really be forbidden. The preconditions for being able to do
`it1 < it2` and `it1 != it2` (or `it1 == it2` for the matter) ought to
be the same. Thus, there's little sense in keeping explicit base object
comparison in operator==() whilst having this is a precondition in
operator<() and operator-() (e.g. used for std::distance() and such).

Co-authored-by: Andrei Golubev <andrey.golubev@intel.com>

dependabot[bot] committed:
Bump cryptography from 42.0.4 to 43.0.1 in /llvm/utils/git (#59)

Bumps [cryptography](https://github.com/pyca/cryptography) from 42.0.4 to 43.0.1.
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/42.0.4...43.0.1)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Maxim-Doronin committed:
Bump cryptography and certifi versions (#65)

ZoranZomborat committed:
Relax negative scales conditions in Quant Dialect (#61)

* relax negative scale conditions
* adapt tests now handling negative scales

sartil committed:
Adding missing check in QuantileQuantizedPerAxisType::verify (#66)

* Adding missing check in QuantileQuantizedPerAxisType::verify
* Adding minor test for illegal quantile array size for per axis type

Maxim-Doronin committed:
Linux precommit (#88)

sartil committed:
Adding new quantileType types to mlir test (#85)

rayngun commented:
RUN_TESTS

rayngun commented:
RUN_TEST

rayngun committed:
GitHub Workflow changes

nikita-kud commented on mlir/test/lib/Dialect/Test/TestDialect.cpp:
```diff
@@ -482,11 +482,11 @@ mlir::LogicalResult test::TestDummyTensorOp::bufferize(
   const auto inputType = getInput().getType();
   const auto bufferizedInputType = test::TestMemrefType::get(
-      getContext(), inputType.getShape(), inputType.getElementType(), nullptr);
+      getContext(), inputType.getShape(), inputType.getElementType(), nullptr, 0);
```
Please add some description on what is the problem and why we need these changes.

I guess this is commit is related to [Customize memref->tensor conversion via dialect interface](https://github.com/intel/npu-plugin-llvm/pull/107/commits/106ade6fd7cef285dd4d842b5e547aa0d15fa0f1) (or [: Replace BaseMemRef/TensorType class with TypeInterface](https://github.com/intel/npu-plugin-llvm/pull/107/commits/dffc016b6ab7f739a786cf83108954b13fd8b340))
Probably it would be better to ammend this commit to keep changes more compact

rayngun replied:
I do end up amending commits. In fact, there are other changes as I cherry-picked each commit. This particular one, I just pushed it right away for testing. I will update soon.

rayngun committed:
Expose TimingManager OutputTextStrategy

VPUx references directly. This was hidden on the LLVM side with
commit 362aa434cc31ccca96749a6db8cd97f5b7d71206

rayngun committed:
Add Value back to SideEffectInterfaces

This reverses partially changed in commit 2c1ae801e1b66a09a15028ae4ba614e0911eec00

dhawalsr committed:
EISW-161791 Adding -D macro to avoid compilation failure of mlir-src-sharder

Signed-off-by: Dhawal Srivastava <dhawal.srivastava@intel.com>

beleiuandrei commented on .github/dependabot.yml:
```diff
@@ -16,3 +16,48 @@ updates:
+  - package-ecosystem: docker
+    directory: /libcxx/utils/ci
+    schedule:
+      interval: daily
```
newline needed.

rayngun committed:
Revert "[mlir][Symbol] Add verification that symbol's parent is a SymbolTable (#80590)"

This reverts commit 7d055af14b7dd7e782b87fb883205eda65e8bd44.

rayngun commented:
RUN_CI_ALL

nikita-kud commented on .github/workflows/linux-precommit.yml:
```diff
@@ -0,0 +1,95 @@
+
+on:
+  push:
+    branches: [ "npu/release/18.x" ]
```
should it be 19.x?

rayngun replied:
Oops. I updated this in the subsequent "github workflow changes" commit. But it is 19.x now.

hrotuna commented on mlir/include/mlir/IR/OperationSupport.h:
```diff
@@ -1187,6 +1189,8 @@ class OpPrintingFlags {
   /// Return if the given ElementsAttr should be printed as hex string.
   bool shouldPrintElementsAttrWithHex(ElementsAttr attr) const;
+  
+  bool allowPrintingElementsAttrAsHex() const;
```
This commit is no longer necessary since it was upstreamed as `shouldPrintElementsAttrWithHex`. It's possible other changes were upstream as well. There is an epic covering this topic: [EISW-101222](https://jira.devtools.intel.com/browse/EISW-101222). But perhaps @andrey-golubev knows better if other commits could be dropped, since he has upstreamed them.

rayngun replied:
Thanks for pointing this out. I originally did not include it due to a compilation issue. I probably should have spent more time on it. I've removed this commit now.

andrey-golubev reviewed (commented):
As @hrotuna mentioned, you can drop from your branch:
* 18b194cc93e6d79dfbb9d59ee41b766da03cb49e

Other feedback:
* if 9cca31c434ca371118c8a3a4b88c7df992869ac9 is a revert, please at least mention the original patch in LLVM
* 0b8b6ff2e9a3d332eb2e51fe76c050487a6163f6 - please reference the patch in LLVM that moved it internally
* d73ce9b7a6ee6ce04e8d4ddebb2bb9662c5245b0 - can we do this in NPU instead? I see `LLVM_ABI_BREAKING_CHECKS` CMake variable - can we set it to ON (or OFF) to achieve the same without modifying LLVM's cmake?
* f54a04769e3c21e28f85ee0c3b268915c6aad8d4 - do we really need it? What did fail in NPU?

rayngun commented:
> As @hrotuna mentioned, you can drop from your branch:
> 
> * [18b194c](https://github.com/intel/npu-plugin-llvm/commit/18b194cc93e6d79dfbb9d59ee41b766da03cb49e)
Ok. Will remove.
> 
> Other feedback:
> 
> * if [9cca31c](https://github.com/intel/npu-plugin-llvm/commit/9cca31c434ca371118c8a3a4b88c7df992869ac9) is a revert, please at least mention the original patch in LLVM
It does not reverse a patch in LLVM completely. I added in original patch in commit comments.
> * [0b8b6ff](https://github.com/intel/npu-plugin-llvm/commit/0b8b6ff2e9a3d332eb2e51fe76c050487a6163f6) - please reference the patch in LLVM that moved it internally
Added commit hash of LLVM patch to commit comment.
> * [d73ce9b](https://github.com/intel/npu-plugin-llvm/commit/d73ce9b7a6ee6ce04e8d4ddebb2bb9662c5245b0) - can we do this in NPU instead? I see `LLVM_ABI_BREAKING_CHECKS` CMake variable - can we set it to ON (or OFF) to achieve the same without modifying LLVM's cmake?
Ok. will look into this
> * [f54a047](https://github.com/intel/npu-plugin-llvm/commit/f54a04769e3c21e28f85ee0c3b268915c6aad8d4) - do we really need it? What did fail in NPU?
See [EISW-163562](https://jira.devtools.intel.com/browse/EISW-163562)

sramasit committed:
Update CMake configuration in Linux precommit workflow to enable custom crash diagnostics
